02.calendar/cal.rb

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env ruby
+require 'date'
+require 'optparse'
+require 'debug'
+
+def main
+  year = Date.today.year
+  month = Date.today.month
+  if ARGV.any?
+    opt = OptionParser.new
+    opt.on("-y [YYYY]", "Specify a year") {|value| year = value.to_i}
+    opt.on("-m [MM]", "Specify a month") {|value| month = value.to_i}
+    opt.parse!
+  end
```

Comment:
>  また、OptionParser用に別のメソッドを定義したかったのですが、方法が見つからず、「main 」に残してしまいました。

どういう形のメソッドを想定していたのかがわかりませんが、たとえばこんなメソッドを用意すれば、

```ruby
def do_something
  year = Date.today.year
  month = Date.today.month
  if ARGV.any?
    opt = OptionParser.new
    opt.on("-y [YYYY]", "Specify a year") {|value| year = value.to_i}
    opt.on("-m [MM]", "Specify a month") {|value| month = value.to_i}
    opt.parse!
  end
  [year, month]
end
```

こんなふうにmainメソッドを変更できます。

```suggestion
  year, month = do_something
```

もしイメージと違っていたらもう少し詳しく教えてください。

Reply:
私のコメントが不明ですみませんでした。 
私が見落としていた部分は、`year、month = do_something`でした。ありがとうございました。
また、`[year, month]`　は2つの変数を返していますが、`return year, month` と比べて何か特別なシンタックスなのでしょうか？

Reply:
> また、[year, month]　は2つの変数を返していますが、return year, month と比べて何か特別なシンタックスなのでしょうか？

意味的にはほぼ同じですね。

`return year, month`は`return [year, month]`と同じ
↓
Rubyはメソッドで最後に評価された式がメソッドの戻り値になるのでreturnは省略できる
↓
つまり`return [year, month]`は`[year, month]`と同じ

という感じです〜。